Mathlib/Probability/CondVar.lean

```diff
@@ -0,0 +1,194 @@
+    Var[X ; μ | m] = fun ω ↦ (X ω - (μ[X | m]) ω) ^ 2 :=
+  condExp_of_stronglyMeasurable _ ((hX.sub stronglyMeasurable_condExp).pow _) hXint
+
+lemma condVar_of_not_integrable (hXint : ¬ Integrable (fun ω ↦ (X ω - (μ[X | m]) ω) ^ 2) μ) :
+    Var[X ; μ | m] = 0 := condExp_of_not_integrable hXint
+
+@[simp]
+lemma condVar_zero : Var[0 ; μ | m] = 0 := by
+  by_cases hm : m ≤ m₀
+  swap; · rw [condVar_of_not_le hm]
+  by_cases hμm : SigmaFinite (μ.trim hm)
+  swap; · rw [condVar_of_not_sigmaFinite hμm]
+  rw [condVar_of_stronglyMeasurable stronglyMeasurable_zero]
+  · simp [← Pi.zero_def]
+  · simpa [-integrable_zero] using integrable_zero ..
+
+@[simp]
+lemma condVar_const (hm : m ≤ m₀) (c : ℝ) : Var[fun _ ↦ c ; μ | m] = 0 := by
+  obtain rfl | hc := eq_or_ne c 0
+  · simp [← Pi.zero_def]
+  by_cases hμm : IsFiniteMeasure μ
+  · simp [condVar, hm, Pi.pow_def]
+  · simp [condVar, condExp_of_not_integrable, integrable_const_iff_isFiniteMeasure hc,
+      integrable_const_iff_isFiniteMeasure <| pow_ne_zero _ hc, hμm, Pi.pow_def]
+
+lemma condVar_ae_eq_condExpL1 [hμm : SigmaFinite (μ.trim hm)] (X : Ω → ℝ) :
+    Var[X ; μ | m] =ᵐ[μ] condExpL1 hm μ ((X - μ[X | m]) ^ 2) :=
+  condExp_ae_eq_condExpL1 ..
+
+lemma condVar_ae_eq_condExpL1CLM [SigmaFinite (μ.trim hm)]
+    (hX : Integrable ((X - μ[X | m]) ^ 2) μ) :
+    Var[X ; μ | m] =ᵐ[μ] condExpL1CLM _ hm μ (hX.toL1 <| (X - μ[X | m]) ^ 2) :=
+  condExp_ae_eq_condExpL1CLM ..
+
+lemma stronglyMeasurable_condVar : StronglyMeasurable[m] (Var[X ; μ | m]) :=
+  stronglyMeasurable_condExp
+
+lemma condVar_congr_ae (h : X =ᵐ[μ] Y) : Var[X ; μ | m] =ᵐ[μ] Var[Y ; μ | m] :=
+  condExp_congr_ae <| by filter_upwards [h, condExp_congr_ae h] with ω hω hω'; dsimp; rw [hω, hω']
+
+lemma condVar_of_aestronglyMeasurable [hμm : SigmaFinite (μ.trim hm)]
+    (hX : AEStronglyMeasurable[m] X μ) (hXint : Integrable ((X - μ[X | m]) ^ 2) μ) :
+    Var[X ; μ | m] =ᵐ[μ] (X - μ[X | m]) ^ 2 :=
+  condExp_of_aestronglyMeasurable' _ ((continuous_pow _).comp_aestronglyMeasurable
+    (hX.sub stronglyMeasurable_condExp.aestronglyMeasurable)) hXint
+
+lemma integrable_condVar : Integrable Var[X ; μ | m] μ := by
+  by_cases hm : m ≤ m₀
+  swap; · rw [condVar_of_not_le hm]; exact integrable_zero _ _ _
+  by_cases hμm : SigmaFinite (μ.trim hm)
+  swap; · rw [condVar_of_not_sigmaFinite hμm]; exact integrable_zero _ _ _
+  haveI : SigmaFinite (μ.trim hm) := hμm
+  exact (integrable_condExpL1 _).congr (condVar_ae_eq_condExpL1 X).symm
+
+/-- The integral of the conditional expectation `μ[X | m]` over an `m`-measurable set is equal to
+the integral of `X` on that set. -/
+lemma setIntegral_condVar [SigmaFinite (μ.trim hm)] (hX : Integrable ((X - μ[X | m]) ^ 2) μ)
+    (hs : MeasurableSet[m] s) :
+    ∫ ω in s, (Var[X ; μ | m]) ω ∂μ = ∫ ω in s, (X ω - (μ[X|m]) ω) ^ 2 ∂μ :=
+  setIntegral_condExp _ hX hs
+
+lemma condVar_ae_eq_condExp_sq_sub_condExp_sq (hm : m ≤ m₀) [IsFiniteMeasure μ] (hX : Memℒp X 2 μ) :
```

Comment:
```suggestion
lemma condVar_ae_eq_condExp_sq_sub_sq_condExp (hm : m ≤ m₀) [IsFiniteMeasure μ] (hX : Memℒp X 2 μ) :
```
?

Reply:
I don't really like that because `^ 2` is postfix, but I guess the current name is a bit silly